app/app/api/private-cloud/analytics/pod-usage-metrics/route.ts

```diff
@@ -0,0 +1,21 @@
+import { z } from 'zod';
+import createApiHandler from '@/core/api-handler';
+import { OkResponse } from '@/core/responses';
+import getPodMetrics from '@/helpers/pod-usage-metrics';
+
+const queryParamSchema = z.object({
+  licencePlate: z.string(),
+  namespacePostfix: z.string(),
+  cluster: z.string(),
+});
+
+const apiHandler = createApiHandler({
+  roles: ['user'],
+  validations: { queryParams: queryParamSchema },
+});
+
+export const GET = apiHandler(async ({ queryParams }) => {
+  const { licencePlate, namespacePostfix, cluster } = queryParams;
```

Comment:
let's have the permission checking here. check if the has the global admin or part of the product team.